### PR TITLE
Tracker Budget App

### PR DESCRIPTION
**Pull request by Jaymelfah**
#### [Video Demo of mobile app](https://www.loom.com/share/c66bc3e6c1964074b02a8e47135ff502)

> Key Highlights
- A user can register and log in, so that the data is private to them.
- A user can introduce new transactions associated with a category.
- A user can see the money spent on each category.
- There is a homepage(categories), transactions page(expenditures) and Add new category page.

### CODE REVIEW

File: app/views/layouts/_nav.html.erb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<nav class="splash w-full flex gap-20 text-white p-6 text-xl">
+  <p><%= link_to '<i class="fa-solid fa-bars"></i>'.html_safe, root_path %></p>
```
Comment (t-yanick) [inline app/views/layouts/_nav.html.erb]: - [x] Great job working on this capstone. Your app is working well, but we have few things to fix as per project requirements.
- The icon you have used for the back button is not quite representive and does not give a good UI for users. The icon you are using for Back is mostly used for menu. Kindly change this icon.
Screen Shot 2023-02-21 at 6 17 18 AM

File: app/views/devise/shared/_links.html.erb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<%- if controller_name != 'sessions' %>
+  <%= link_to "Log in", new_session_path(resource_name) %><br />
+<% end %>
+
+<%- if devise_mapping.registerable? && controller_name != 'registrations' %>
+  <%= link_to "Sign up", new_registration_path(resource_name) %><br />
```
Comment (t-yanick) [inline app/views/devise/shared/_links.html.erb:6]: - [ ] As per project requirements, when a user clicks on the Sign up/Log in button, they are supposed to be redirected to the Categories page. At the moment your app leads to a page with two buttons: `Categories` and `Logout`. I understand you are trying to make the app look beautiful, but at the momemt, you are expected to follow the project requirements.
Screen Shot 2023-02-21 at 6 25 41 AM
Kindly delete the route that leads to this page and redirect both sucessful sign up and login to the categories page which should be the index page.
- See project requirement below after logging in
Screen Shot 2023-02-21 at 6 27 51 AM

File: config/routes.rb [hunk 3]
```diff
@@ -1,6 +1,7 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
+  devise_for :users
+  root 'splash_screen#index'
```
Comment (t-yanick) [inline config/routes.rb]: - [x] Kindly make the categories page your index page

File: app/controllers/categories_controller.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    @category = Category.find(params[:id])
+    @category.destroy
+    flash[:notice] = 'The category was successfully deleted'
+    redirect_to categories_path
+  end
+
+  def category_params
+    params.require(:category).permit(:icon, :name)
+  end
+end
```
Comment (t-yanick) [inline app/controllers/categories_controller.rb:45]: - [ ] I think the category_params should be a private method. Kindly review this.

File: app/controllers/expenditures_controller.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    @category = Category.find(params[:category_id])
+    @expenditure = Expenditure.find(params[:id])
+    @expenditure.destroy
+    flash[:notice] = 'Expenditure deleted successfully'
+    redirect_to category_path(@category.id)
+  end
+
+  def expenditure_params
+    params.require(:expenditure).permit(:name, :amount)
+  end
```
Comment (t-yanick) [inline app/controllers/expenditures_controller.rb:42]: - [ ] I think the expenditure_params should be a private method. Kindly review this.

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️

Hi @Jaymelfah , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good video presentation
:heavy_check_mark: Documentation is very explicit
:heavy_check_mark: Spec tests added and all passing
:heavy_check_mark: Design well followed
:heavy_check_mark: All pages present and functionalities as expected
:heavy_check_mark: Database schema reflects E-R diagram.

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/layouts/_nav.html.erb [hunk 6]
```diff
@@ -1,6 +1,8 @@
-<nav class="splash w-full flex gap-20 text-white p-6 text-xl">
-  <p><%= link_to '<i class="fa-solid fa-bars"></i>'.html_safe, root_path %></p>
+<nav class="splash w-full flex gap-12 text-white p-6 text-xl">
+  <p><%= link_to '<i class="fa-solid fa-bars"></i>'.html_safe, unauthenticated_root_path %></p>
```
Comment (Lucas-Masaba) [inline app/views/layouts/_nav.html.erb:2]: - [x] The changes you made here are not being seen on the live version of your project. It still uses the hamburger button as the back button as seen in the image below. You could also evenly space the buttons and the heading Transaction
image

File: config/routes.rb [hunk 7]
```diff
@@ -1,6 +1,15 @@
 Rails.application.routes.draw do
   devise_for :users
-  root 'splash_screen#index'
+
+  devise_scope :user do
+    authenticated :user do
+      root 'categories#index', as: :authenticated_root
```
Comment (Lucas-Masaba) [inline config/routes.rb:6]: - [x] Here you made the home page for categories the root but that is not the case for the live version. Instead, I go to the page with a button for categories and log out. 
image

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @Jaymelfah  👋 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
almost-there

 ## Highlights 🍾 
- Linter checks passing ✔️ 
- Made changes as requested by the previous reviewer 👍 

 ## Required Changes ♻️
- [ ] Please make sure to push the changes you made on your local machine to the live demo.

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — APPROVED** [review]
## Hi @Jaymelfah,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- Amazing design ✔️
- Professional readme file ✔️
209362895-9762f2f8-13cf-4ce0-90b6-a672c4f04196

# Status: Approved ✔️✔️✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [ ] This Project should use git-flow. Your pull request should compare the feature branch and development branch. remember this in your next pull request. Please check again this [git-flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md) link.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**